contracts/libraries/CurveHelpers.sol

```diff
@@ -0,0 +1,155 @@
+    function removeLiquidityOneCoin(
+        ICurvePool pool,
+        uint256 amount,
+        address outputToken,
+        uint256 poolCoinAmount,
+        string memory signature
+    ) internal {
+        bool success;
+        if (poolCoinAmount == 2) {
+            // Curve 2pool
+            if (outputToken == pool.coins(0)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 0, 0));
+            } else {
+                require(outputToken == pool.coins(1), "CH: INVALID_OUTPUT_TOKEN");
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 1, 0));
+            }
+        } else if (poolCoinAmount == 3) {
+            // Curve 3pool
+            if (outputToken == pool.coins(0)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 0, 0));
+            } else if (outputToken == pool.coins(1)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 1, 0));
+            } else {
+                require(outputToken == pool.coins(2), "CH: INVALID_OUTPUT_TOKEN");
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 2, 0));
+            }
+        } else {
+            // Curve 4pool
+            if (outputToken == pool.coins(0)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 0, 0));
+            } else if (outputToken == pool.coins(1)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 1, 0));
+            } else if (outputToken == pool.coins(2)) {
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 2, 0));
+            } else {
+                require(outputToken == pool.coins(3), "CH: INVALID_OUTPUT_TOKEN");
+                (success, ) = address(pool).call(abi.encodeWithSignature(signature, amount, 3, 0));
+            }
+        }
+    }
+
+    /// @dev Get the arrays of obtained token and spent token
+    /// @param inputToken The token spent
+    /// @param inputTokenBalanceBefore The input token balance before
+    /// @param expectedInputAmount The expected amount of input token spent
+    /// @param outputToken The token obtained
+    /// @param outputTokenBalanceBefore The output token balance before
+    /// @param minAmountOut The minimum of output token expected
+    function getOutputAmounts(
+        IERC20 inputToken,
+        uint256 inputTokenBalanceBefore,
+        uint256 expectedInputAmount,
+        IERC20 outputToken,
+        uint256 outputTokenBalanceBefore,
+        uint256 minAmountOut
+    ) internal view returns (uint256[] memory amounts, address[] memory tokens) {
+        uint256 vaultAmount = inputTokenBalanceBefore - inputToken.balanceOf(address(this));
+        require(vaultAmount == expectedInputAmount, "CH: INVALID_AMOUNT_WITHDRAWED");
```

Comment:
Use directly `inputTokenBalanceBefore - inputToken.balanceOf(address(this))` inside the `require`.